idaes/power_generation/carbon_capture/mea_solvent_system/unit_models/column.py

```diff
@@ -0,0 +1,1799 @@
+                "material_balances",
+                "material_flow_linking_constraints",
+                    "material_flow_dx_disc_eq"]:
+                c.activate()
+
+        # solve for a small lenght if stripper
+        if (blk.config.process_type == 'Stripper' or
+                blk.config.process_type == 'Regenerator'):
+            specified_length = value(blk.length_col)
+            blk.length_col.fix(0.6)
+
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+        init_log.info(
+            "Step 2: {}.".format(idaeslog.condition(res)))
+
+        # ---------------------------------------------------------------------
+        print('==============================================================')
+        print('STEP 3: Add Mass tranfer terms')
+        print('        -->(3a) velocities & Interface pressure')
+        print('        -->(3b) Enhancement factor')
+
+        # Initialize :
+        #   --> Velocities
+        #   --> Interface pressure
+        #   --> Enhancement factor
+
+        # velocity
+        blk.vel_vap.unfix()
+        blk.vel_liq.unfix()
+        blk.eq_vel_vap.activate()
+        blk.eq_vel_liq.activate()
+        for t in blk.t:
+            for x in blk.vapor_phase.length_domain:
+                blk.vel_vap[t, x].value = value(
+                    blk.vapor_phase.properties[t, x].flow_mol /
+                    (blk.area_col * blk.vapor_phase.properties[t, x].conc_mol))
+            for x in blk.liquid_phase.length_domain:
+                blk.vel_liq[t, x].value = value(
+                    blk.liquid_phase.properties[t, x].flow_mol /
+                    (blk.area_col * blk.liquid_phase.properties[t, x].conc_mol))
+
+        # Interface pressure
+        blk.pressure_equil.unfix()
+        blk.pressure_at_interface.activate()
+        blk.E.fix(1)
+
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+        init_log.info(
+            "Step 3a: {}.".format(idaeslog.condition(res)))
+        # ----------------------------------------------------------------------
+        # Enhancement factor model
+        blk.E.unfix()
+        for t in blk.t:
+            for x in blk.liquid_phase.length_domain:
+                blk.E[t, x].value = 100
+        blk.yi_MEA.unfix()
+        blk.yeq_CO2.unfix()
+        blk.E1_eqn.activate()
+        blk.E2_eqn.activate()
+        blk.yeq_CO2_eqn.activate()
+
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+        init_log.info("Step 3 complete: {}.".format(idaeslog.condition(res)))
+        # ---------------------------------------------------------------------
+
+        print('============================================================')
+        print('STEP 4: Isothermal chemical absoption')
+        print("Homotopy steps: ")
```

Comment:
Try using the IDAES homotopy solver for this (idaes.core.util.homotopy).

Reply:
I have used the IDAES homotopy solver and it works pretty well. However, it does not give the user the option to control the output to stdout. An option that shows only the solver termination condition and values of the fixed variables at each step is desired.